cmd/config-v16.go

```diff
@@ -131,15 +131,15 @@ func loadConfig(envParams envParams) error {
 
 	// If env is set override the credentials from config file.
 	if globalIsEnvCreds {
-		srvCfg.SetCredential(envParams.creds)
+		srvCfg.SetCredential(globalActiveCred)
+	} else {
+		globalActiveCred = srvCfg.GetCredential()
```

Comment:
> This makes a little confusion about whether we should use globalActiveCred or srvCfg.GetCredential() later. I think we should reduce global variables instead if possible.

All functions should use `globalActiveCred` which is the right credential even though config file has different or same credential.  ATM I follow existing logic of having configuration in sync with env ie `config` struct contains replaced credentials values on load, so there is no change in other functions/method.  Eventually they will be using `globalActiveCred`.  My actual plan is to remove getter/setter of `config` structure wherever possible in down the line.

Spilling env or command line arguments information across code base is not a good idea.

I am favor of removing all globals. I already have a patch ready but the patch is huge and reviewing would be harder.  I split the patch into multiples to ease out for acceptance.  This PR is one of such kind.